Assets/Scripts/Terrain/HeightMap/MultiOctaveNoiseMap.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Creates a height map using multiple octaves of noise generated with various settings.
+/// 
+/// Uses an OctaveNoise gen that uses a PerlinNoise for individual octaves.
+/// 
+/// Using Perlin Noise function from adrian's soapbox "Understanding Perlin Noise" by Flafla2.
+/// https://flafla2.github.io/2014/08/09/perlinnoise.html
+/// From August 9th, 2014
+/// 
+/// github link: https://gist.github.com/Flafla2/f0260a861be0ebdeef76
+/// </summary>
+public class MultiOctaveNoiseMap : AbstractHeightMapGenerator
+{
+    /// <summary>
+    /// Number of different noise octaves to use
+    /// </summary>
+    [Range(1, 16)]
```

Comment:
Consider adding an error message here, https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.rangeattribute?view=netframework-4.8

Reply:
Would it be good to add this to any attributes that have a strict range?

Reply:
Oh wait, that is actually a RangeAtribute from Unity, it doesn't support error messages. https://docs.unity3d.com/ScriptReference/RangeAttribute.html 
It is used to be able to edit attributes easily from the inspector as a slider :)